test/integration/shopping-cart.repository.integration.ts

```diff
@@ -0,0 +1,101 @@
+// Copyright IBM Corp. 2018. All Rights Reserved.
+// Node module: @loopback/example-shopping
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {ShoppingCartRepository} from '../../src/repositories/shopping-cart.repository';
+import {ShoppingCart} from '../../src/models';
+import {expect} from '@loopback/testlab';
+import Docker = require('dockerode');
+
+describe('ShoppingCart KeyValue Repository', () => {
+  let repo: ShoppingCartRepository;
+  let cart1: ShoppingCart;
+  let cart2: ShoppingCart;
+  let docker: Docker;
+  let container: Docker.Container;
+
+  before(async function() {
+    // tslint:disable-next-line:no-invalid-this
+    this.timeout(0);
+    docker = new Docker();
+    container = await docker.createContainer({
+      Image: 'redis',
+      Tty: true,
+      ExposedPorts: {
+        '6379/tcp': {},
+      },
+      HostConfig: {
+        PortBindings: {
+          '6379/tcp': [
+            {
+              HostPort: '6379',
+            },
+          ],
+        },
+      },
+    });
+    await container.start();
```

Comment:
How long does it take to start a Redis container?

I am concerned about the overhead this may be adding to `npm t`. When developing locally, we don't want to (re)start the Redis instance every time we want to re-run our test suite. 

Instead, we should start any database instances beforehand. When running on CI, the CI build script should start the instances before running the tests - Travis CI provides hooks for that.

See also https://github.com/strongloop/loopback4-example-shopping/pull/5. Since Travis CI already supports Redis, I think we should use the same approach from #5 and use Travis-CI provided instance of Redist instead of using Docker.

Related best practice: [Clean the database before each test](https://loopback.io/doc/en/lb4/Testing-your-application.html#clean-the-database-before-each-test).